client/document/index.jsx

```diff
@@ -173,13 +174,37 @@ class Document extends React.Component {
 						} }
 					/>
 
+					{ // Use <script nomodule> to redirect browsers with no ES module
+					// support to the fallback build. ES module support is a convenient
+					// test to determine that a browser is modern enough to handle
+					// the evergreen bundle.
+					addEvergreenCheck && (
+						<script
+							nonce={ inlineScriptNonce }
+							noModule
+							dangerouslySetInnerHTML={ {
+								__html: `
+							(function() {
+								var url = window.location.href;
+
+								if ( url.indexOf( 'forceFallback=1' ) === -1 ) {
+									url += ( url.indexOf( '?' ) !== -1 ? '&' : '?' );
+									url += 'forceFallback=1';
+									window.location.href = url;
```

Comment:
Does this mean that fallback users who use IE11 et al will wee `forceFallback` in their URLs all the time?

Reply:
Could the `forceFallback` be just initial way to do the switch and next page-loads would pick it up from `localStorage`?

Reply:
No, `forceFallback` will only be used if somehow we provide an evergreen bundle to a non-evergreen browser.

This is probably not super well explained in the PR, so I'll explain it here in more detail. The primary method of determining which build to use is the user agent string. If the user agent string for one of the known evergreen browsers is detected, then the browser gets sent the evergreen build. Otherwise it gets sent the fallback build.

This line here is a secondary mechanism designed to avoid breakage. It checks to see if the current browser supports ES modules (a good litmus test for browser "modernness" at this point in time), and if it doesn't, redirects the user. Now, this should only happen if the browser misreports its user agent string (e.g. by impersonating a newer browser), but since it leads to breakage, it made sense in my mind to add this escape hatch.

If we're happy relying on user agent strings alone, we can remove this check here and even remove `forceFallback` altogether (although it arguably could be left behind as a "hidden feature" for solving potential compatibility issues).

Reply:
Thanks for the explanation. If it's just an emergency detection mechanism rather than the primary one, ugly URLs are OK and it's worth having it.